modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/service/test/SegmentsExperimentLocalServiceTest.java

```diff
@@ -674,43 +688,140 @@ public void testUpdateSegmentsExperimentNameInStatusRunning()
 	}
 
 	@Test
-	public void testUpdateSegmentsExperimentStatusToCompletedWithWinnerSegmentsExperience()
+	public void testUpdateSegmentsExperimentStatusToCompletedWithWinnerSegmentsExperienceAndControlSegmentsExperienceWithPriorityMinusTwo()
 		throws Exception {
 
-		SegmentsExperience segmentsExperience1 = _addSegmentsExperience();
-
-		SegmentsExperience segmentsExperience2 =
-			SegmentsTestUtil.addSegmentsExperience(
-				segmentsExperience1.getGroupId(),
-				segmentsExperience1.getClassNameId(),
-				segmentsExperience1.getClassPK());
+		SegmentsExperience segmentsExperience1 = _addSegmentsExperience(0);
+		SegmentsExperience segmentsExperience2 = _addSegmentsExperience(-2);
+		SegmentsExperience segmentsExperience3 = _addSegmentsExperience(-3);
 
 		SegmentsExperiment segmentsExperiment = _addSegmentsExperiment(
-			segmentsExperience1);
+			segmentsExperience2);
 
 		SegmentsExperience variantSegmentsExperience =
-			SegmentsTestUtil.addSegmentsExperience(
-				segmentsExperience1.getGroupId(),
-				segmentsExperience1.getClassNameId(),
-				segmentsExperience1.getClassPK());
+			_addVariantSegmentsExperience(
+				segmentsExperience2.getSegmentsEntryId(),
+				segmentsExperiment.getSegmentsExperimentId());
 
-		_segmentsExperienceLocalService.updateSegmentsExperienceActive(
-			variantSegmentsExperience.getSegmentsExperienceId(), false);
+		segmentsExperiment.setStatus(
+			SegmentsExperimentConstants.STATUS_FINISHED_WINNER);
 
-		_segmentsExperimentRelLocalService.addSegmentsExperimentRel(
-			segmentsExperiment.getSegmentsExperimentId(),
-			variantSegmentsExperience.getSegmentsExperienceId(),
-			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+		_segmentsExperimentLocalService.updateSegmentsExperiment(
+			segmentsExperiment);
 
 		_segmentsExperimentLocalService.updateSegmentsExperimentStatus(
 			segmentsExperiment.getSegmentsExperimentId(),
-			SegmentsExperimentConstants.STATUS_RUNNING);
+			variantSegmentsExperience.getSegmentsExperienceId(),
+			SegmentsExperimentConstants.STATUS_COMPLETED);
+
+		segmentsExperience1 =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				segmentsExperience1.getSegmentsExperienceId());
+
+		Assert.assertTrue(segmentsExperience1.isActive());
+		Assert.assertEquals(0, segmentsExperience1.getPriority());
```

Comment:
Better if you use DEFAULT_PRIORITY - X , as you recommended my in the previous revision.

Reply:
you meant just here? or in every place I'm using the priority?

Reply:
In every place we use the priority.

Reply:
Done ;)